### PR TITLE
fix: Fix integration of extra kernel args

### PR DESCRIPTION
**Pull request by bradbeam**
Signed-off-by: Brad Beam <brad.beam@talos-systems.com>

### CODE REVIEW

File: cmd/osctl/cmd/install_linux.go [hunk 1]
```diff
@@ -91,6 +91,6 @@ func init() {
 	installCmd.Flags().StringVar(&device, "device", "", "The path to the device to install to")
 	installCmd.Flags().StringVar(&endpoint, "userdata", "", "The value of "+constants.KernelParamUserData)
 	installCmd.Flags().StringVar(&platform, "platform", "", "The value of "+constants.KernelParamPlatform)
-	installCmd.Flags().StringSliceVar(&extraKernelArgs, "extra-kernel-arg", []string{}, "Extra argument to pass to the kernel")
+	installCmd.Flags().StringArrayVar(&extraKernelArgs, "extra-kernel-arg", []string{}, "Extra argument to pass to the kernel")
```
Comment (bradbeam) [inline cmd/osctl/cmd/install_linux.go]: When using a `StringSliceVar`, it splits on a comma. This breaks settings that include a comma such a baud rate ( `ttyS1,115200n8` ). Switching to `StringArrayVar` keeps the comma intact.

File: internal/pkg/kernel/kernel.go [hunk 2]
```diff
@@ -20,10 +20,10 @@ func NewDefaultCmdline() *cmdline {
 	cmdline.Append("pti", "on")
 	// TODO(andrewrynhard): Add slub_debug=P. See https://github.com/talos-systems/talos/pull/157.
 	cmdline.Append("consoleblank", "0")
-	cmdline.Append("console", "ttyS1,115200n8")
+	//cmdline.Append("console", "ttyS1,115200n8")
```
Comment (bradbeam) [inline internal/pkg/kernel/kernel.go]: Leaving this comment in for discussion. I'm not sure of the best place to migrate these parameters to.

Comment (bradbeam) [reply internal/pkg/kernel/kernel.go]: These are currently called inline:

```
internal//app/init/internal/platform/iso/iso.go:	cmdline := kernel.NewDefaultCmdline()
internal//app/init/internal/platform/baremetal/baremetal.go:	cmdline := kernel.NewDefaultCmdline()
internal//app/init/internal/platform/cloud/packet/packet.go:	cmdline := kernel.NewDefaultCmdline()
```

Comment (andrewrynhard) [reply internal/pkg/kernel/kernel.go]: I wonder if we need `--append-kernel-arg` and `--set-kernel-arg`. Then use `cmdline.Set(...)`. 🤔 not the prettiest. Or somehow in the `NewDefaultCmdline` we handle this.

File: hack/installer/entrypoint.sh [hunk 3]
```diff
@@ -86,14 +87,17 @@ case "$1" in
...
+          ;;
         n )
           TALOS_RAW="/out/${OPTARG}.raw"
           ;;
         p )
           TALOS_PLATFORM=${OPTARG}
           echo "Using kernel parameter talos.platform=${TALOS_PLATFORM}"
           ;;
-       r )
+        r )
```
Comment (smira) [inline hack/installer/entrypoint.sh]: extra space?

File: internal/pkg/kernel/kernel.go (hunk 2 above)
Comment (andrewrynhard) [reply internal/pkg/kernel/kernel.go]: The quick win is that we just be explicit in our image creation (i.e. just use `-e` in the `Makefile`).

File: hack/installer/entrypoint.sh (hunk 3 above)
Comment (andrewrynhard) [reply hack/installer/entrypoint.sh]: I think this just aligns with the other cases.

Comment (bradbeam) [reply hack/installer/entrypoint.sh]: ^ yeah just a little alignment fix

**Pull request merged**